File: ex/sucess/spojWILLITST.py
n = input()
m = set()
while n > 1 and n not in m:
    m.add(n)
    if n % 2 == 0:
        n = n / 2
    else:
        n = 3*n + 3
if n > 1: print 'NIE'
else: print 'TAK'

File: ex/sucess/spojWILLITST.cpp
#include <iostream>
#include <set>
using namespace std;

int main(int argc, char const *argv[]) {
    unsigned long long n;
    set<unsigned long long> m;

    for (cin >> n; n > 1 && !m.count(n); m.insert(n), n = (n % 2 == 0)? n/2 : 3*n+3);

    if (n > 1) cout << "NIE" << endl;
    else cout << "TAK" << endl;

    return 0;
}
